node/actors/internal_nats_server.go

```diff
@@ -95,8 +156,182 @@ func (ns *internalNatsServer) buildAgentCredentials() ([]agentCredential, error)
 	return creds, nil
 }
 
-func (ns *internalNatsServer) startNatsServer(creds []agentCredential) error {
+func (ns *internalNatsServer) startNatsServer(opts *server.Options) error {
 	ns.Log().Debug("Starting internal NATS server")
-	// TODO!
+
+	s, err := server.NewServer(opts)
+	if err != nil {
+		server.PrintAndDie("nats-server: " + err.Error())
+		return err
+	}
+
+	// FIXME-- read from config
+	// if debug || trace {
+	// 	s.ConfigureLogger()
+	// }
+
+	if err := server.Run(s); err != nil {
+		server.PrintAndDie("nats-server: " + err.Error())
+		return err
+	}
+
 	return nil
 }
+
+func (ns *internalNatsServer) generateConfig() (*server.Options, error) {
+	hostPub, err := ns.hostUser.PublicKey()
+	if err != nil {
+		return nil, err
+	}
+
+	hostSeed, err := ns.hostUser.Seed()
+	if err != nil {
+		return nil, err
+	}
+
+	data := &configTemplateData{
+		Credentials:       make(map[string]*credentials),
+		Connections:       make(map[string]*nats.Conn),
+		NexHostUserPublic: hostPub,
+		NexHostUserSeed:   string(hostSeed),
+	}
+
+	for _, cred := range ns.creds {
+		seed, err := cred.nkey.Seed()
+		if err != nil {
+			return nil, err
+		}
+
+		pubkey, err := cred.nkey.PublicKey()
+		if err != nil {
+			return nil, err
+		}
+
+		data.Credentials[cred.workloadType] = &credentials{
+			WorkloadType: cred.workloadType,
+			NkeySeed:     string(seed),
+			NkeyPublic:   pubkey,
+		}
+	}
+
+	bytes, err := ns.generateTemplate(data)
+	if err != nil {
+		ns.Log().Error("Failed to generate internal nats server config file", slog.Any("error", err))
+		return nil, err
+	}
+
+	opts := &server.Options{
+		JetStream: true,
+		StoreDir:  filepath.Join(os.TempDir(), defaultInternalNatsStoreDir), // FIXME-- use unique temp location
+		Port:      -1,
+		// Debug:     debug, FIXME-- make configurable
+		// Trace:     trace,
+	}
+
+	f, err := os.CreateTemp(os.TempDir(), defaultInternalNatsConfigFile)
+	if err != nil {
+		return nil, err
+	}
+	defer os.Remove(f.Name()) // clean up
+
+	if _, err := f.Write(bytes); err != nil {
+		ns.Log().Error("Failed to write internal nats server config file", slog.Any("error", err))
+		return nil, err
+	}
+
+	err = opts.ProcessConfigFile(f.Name())
+	if err != nil {
+		ns.Log().Error("Failed to process configuration file", slog.Any("error", err))
+		return nil, err
+	}
+
+	return opts, nil
+}
+
+func (ns *internalNatsServer) generateTemplate(config *configTemplateData) ([]byte, error) {
+	var wr bytes.Buffer
+
+	t := template.Must(template.New("natsconfig").Parse(configTemplate))
+	err := t.Execute(&wr, config)
+	if err != nil {
+		return nil, err
+	}
+
+	// -8 is equivalent to TRACE-ish level
+	// FIXME... .Log() does not exist
+	// ns.Log().Log(context.Background(), -8, "generated NATS config", ns.Log().String("config", wr.String()))
```

Comment:
ns.Log() does indeed exist. We should be able to use ns.Log().LevelFunc(...,) here.

Reply:
but we definitely don't need to log the whole config file output.

Reply:
It doesn't map nicely to slog levels. We have a fix for this in #389.

Reply:
it mapped identically before. I was able to use the slog level as input. Instead of making a big switch statement and then making explicit `Error` and `Info` calls, why not create a simple mapping function that converts the ergo log level to a slog log level?

Reply:
Can you add this feedback to #389?